### DISCUSSION TITLE
How can I tell if someone has played an audio element?

### DISCUSSION THREAD

**Discussion by benjaminsnorris** [plausible/analytics#2164] [Anything Else]
I'm experimenting with adding an audio version of my blog posts, and want to find out if anyone uses it. Is there a way that I can know if a visitor engages with the audio element? I was thinking I could set up a goal to track that more easily, but I don't see a way that I could hook into when the visitor clicks play.

I saw #751 about engagement stats, but I'm just wondering if there is any way to know if it's used, even if I can't yet get robust stats. Thanks.

**Comment by metmarkosaric** [comment]
hi @benjaminsnorris! we don't have any automatic tracking for audio/video plays but you can set up a custom event for these clicks. that will give you the insights into how many people click on the play button etc. see https://plausible.io/docs/custom-event-goals

**Reply by benjaminsnorris** [reply]
Thanks @metmarkosaric. I looked through that doc, but couldn't tell how to tie a javascript event to someone interacting with the `audio` element.

My workaround for right now will be to hide the `audio` element, and instead show a button and fire javascript when that button is clicked that will hide the button, show the audio player, and fire the goal. I'd love to not have to do the workaround.

**Comment by RobertJoonas** [comment]
I think with a `HTMLMediaElement` (either `<video>` or `<audio>`) you should be able to do this (example only with `audio,` code should be identical with `video`):

```html
<!-- Give the audio element an ID to find it with JavaScript -->
<audio id="trackedAudio" controls src="your_audio.mp3">
    Your browser doesn't support this audio element
</audio>

<script>
    var audioElement = document.getElementById('trackedAudio')
    
    // Set a flag to ignore the case where playing has already started and is resumed from paused state
    var audioAlreadyStarted = false

    // Listen to the 'play' event on the <audio> element. Ingore if already started playing and not finished yet
    audioElement.addEventListener('play', function (e) {
        if (!audioAlreadyStarted) {
            audioAlreadyStarted = true
            plausible('Audio Play')
        }
    })

    // Listen to the 'ended' event on the <audio> element to reset the flag and start listening again
    audioElement.addEventListener('ended', function (e) {
        audioAlreadyStarted = false
    })
</script>
```

Do note that the JavaScript code in the `<script>` tag should come **after** the media element in your HTML code. Safe to put the script just before the closing `</body>` tag.

@benjaminsnorris, please do let us know if this works out for you. Would make sense to add this to the docs too, I think. Hope this helps, thanks!

**Reply by benjaminsnorris** [reply]
That looks great! Thank you so much. I agree that this would make a good addition to the docs.
